src/js/components/conversation.jsx

```diff
@@ -62,7 +62,32 @@ export class ConversationComponent extends Component {
     }
 
     render() {
-        const messages = this.props.conversation.messages.map((message) => {
+        const processedMessages = [...this.props.conversation.messages];
+        let lastAuthor;
+        processedMessages.forEach((message, index) => {
+            const author = message.role === 'appUser' ? message.role : `${message.role}_${message.name}`;
+
+            if (!lastAuthor) {
+                lastAuthor = author;
+                message.firstInGroup = true;
```

Comment:
That could even be an object instead with the message id as the key so you don't have to worry about sorting and the look would be quicker